lib/Error/Base.php

```diff
@@ -4,66 +4,75 @@
 
 use Exception;
 
+/**
+ * Base is the base error from which all other more specific Stripe errors derive.
+ *
+ * @package Stripe\Error
+ */
 abstract class Base extends Exception
 {
+    protected $httpBody;
+    protected $httpHeaders;
+    protected $httpStatus;
+    protected $jsonBody;
+    protected $requestId;
+    protected $stripeCode;
+
     public function __construct(
         $message,
         $httpStatus = null,
         $httpBody = null,
         $jsonBody = null,
-        $httpHeaders = null
+        $httpHeaders = null,
+        $stripeCode = null
```

Comment:
Any particular reason to call this "stripe code" here, and just "code" above? Should we just call it "code" to be consistent with what's in the API reference/JSON structure?

Reply:
We can't call it `code` because the `Exception` class already has an integer attribute (and accessors) with that name: https://www.php.net/manual/en/class.exception.php

Reply:
Fuuuu. Okay, I don't have too strong of opinion here, but given we can't call it "code" anyway and OAuth errors already have an "error code", should we call it `$errorCode`/`$getErrorCode` instead?

The actual JSON object fields are named differently (`code` in normal errors versus `error` in OAuth errors I think), but it probably wouldn't hurt just to have the naming the same in PHP. `OAuthBase` extends `Error\Base`, but you could probably promote the existing `$errorCode` up to the parent.

Again, no strong opinion, but just wanted to make sure that you considered the possibility.

Reply:
So my plan is to do something similar to https://github.com/stripe/stripe-ruby/pull/811 as a followup to this PR, i.e. expose the Stripe error "API resource" on the exception via `getError()` or something similar. On the API resource, we won't have this naming constraint so people will be able to do `$e->getError()->getCode()`.

On the exception object itself, I think we might as well keep the current name to avoid an unnecessary breaking change. wdyt? (Note that this will be released in a new major version, so breaking changes are an option if we think it's for the best.)

Reply:
> So my plan is to do something similar to https://github.com/stripe/stripe-ruby/pull/811 as a followup to this PR, i.e. expose the Stripe error "API resource" on the exception via `getError()` or something similar. On the API resource, we won't have this naming constraint so people will be able to do `$e->getError()->getCode()`.

Ah, I see! Yeah, that sounds great.

> On the exception object itself, I think we might as well keep the current name to avoid an unnecessary breaking change. wdyt? (Note that this will be released in a new major version, so breaking changes are an option if we think it's for the best.)

Oh man, there was enough changes in that file that I didn't even notice that `$stripeCode` /`$getStripeCode` existed before — I thought it was new. +1 that a breaking change isn't worth it. Ship away!